assets/js/imdbae/chat/room.js

```diff
@@ -0,0 +1,71 @@
+//Attrib: Let's build a slack clone from Medium
+
+
+const initialState = {
+	channel: null,
+	currentRoom: {}, 
+	messages: [],
+	presentUsers: [],
+	loadingOlderMessages: false, 
+	pagination: {
+		total_pages: 0,
+		total_entries: 0,
+		page_size: 0,
+		page_number: 0,
+	},
+}
+
+export default function (state = initialState, action) {
+	switch(action.type) {
+		case 'ROOM_CONNECTED_TO_CHANNEL':
+			return {
+				...state,
+				channel: action.channel,
+				currentRoom: action.response.room,
+				messages: action.response.messages.reverse(),
+				pagination: action.response.pagination,
+			};
+		case 'USER_LEFT_ROOM':
+			return initialState;
+		case 'MESSAGE_CREATED':
+			return {
+				...state,
+				messages: [
+					...state.messages, 
+					action.message, 
```

Comment:
Based on your dispatch call on chatbox.jsx (`dispatch({type: 'MESSAGE_CREATED', payload: message})`), I think this should be `action.payload`